os/root-on-raid.md

```diff
@@ -0,0 +1,58 @@
+# Setting up Root on Raid
+
+Partitions used in RAID arrays need to have the gpt partition type uuid `be9067b9-ea49-4f15-b4f6-f36f8c9e1818` in order to boot. Because Ignition does not current support modifying existing partitions
+without wiping the partition table, the default ROOT partition cannot be used since it has different partition type uuid. In order to have the root partition in a raid array:
+ * Create the partitions used in the RAID array with the gpt partition type uuid `be9067b9-ea49-4f15-b4f6-f36f8c9e1818`
+ * Create a RAID array with the created partitions 
+ * Create a filesystem with on the raid array with the ROOT label
+ * Remove the ROOT label from the old root filesystem by wiping the existing filesystem or overwriting it
+
+## Example CLC
+
+```yaml container-linux-config
+storage:
+  disks:
+    - device: /dev/vdb
+      wipe_table: true
+      partitions:
+       - label: root1
+         number: 1
+         size: 256MiB
+         start: 0
+         type_guid: be9067b9-ea49-4f15-b4f6-f36f8c9e1818
+       - label: root2
+         number: 2
+         size: 256MiB
+         start: 0
+         type_guid: be9067b9-ea49-4f15-b4f6-f36f8c9e1818
+  raid:
+    - name: "my_raid_array"
+      level: "raid1"
+      devices:
+        - "/dev/vdb1"
+        - "/dev/vdb2"
+  filesystems:
+    - name: "ROOT"
+      mount:
+        device: "/dev/md/my_raid_array"
+        format: "ext4"
+        create:
+          options:
+            - "-L"
+            - "ROOT"
+    - name: "unused"
+      mount:
+        device: "/dev/vda9"
+        format: "ext4"
+        create:
+          options:
+            - "-L"
+            - "unused"
+          force: true
+```
+
+## Under the Hood
+Since Container Linux has an immutable initramfs, the initramfs must have a predetermined way for finding the root filesystem. If the root partition is in a RAID array, the raid array must be started in order to find the root filesystem. To avoid assembling unnecessary arrays (which may be assembled with other options from mdadm.conf), the partitions that are part of the root raid array must have the partition type uuid `be9067b9-ea49-4f15-b4f6-f36f8c9e1818`. A udev rule in the initramfs autoassembles all partitions with that type uuid.
+
+## Future
+Ignition will support resizing and relabling partitions which allow the root partition to be resized and assigned a new type uuid. This will allow it to be reused as part of a raid array.
```

Comment:
There might be some value to this, but only in a particular context: if we've explained the situation well enough that an informed user is now thinking "that's odd", then we can fill in details about why we're doing things that way and how we're fixing the process.  But until we're at that level of clarity, talking about future changes will only confuse matters further.